backend/raw/raw.go

```diff
@@ -0,0 +1,127 @@
+package raw
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/diskfs/go-diskfs/backend"
+)
+
+type rawBackend struct {
+	storage  fs.File
+	readonly bool
+}
+
+// Create a backend.Storage from provided fs.File
+func New(f fs.File, isReadonly bool) backend.Storage {
+	return rawBackend{
+		storage:  f,
+		readonly: isReadonly,
+	}
+}
+
+// Create a backend.Storage from a path to a device
+// Should pass a path to a block device e.g. /dev/sda or a path to a file /tmp/foo.img
+// The provided device/file must exist at the time you call OpenFromPath()
+func OpenFromPath(pathName string, isReadonly bool) (backend.Storage, error) {
+	if pathName == "" {
+		return nil, errors.New("must pass device of file name")
+	}
+
+	if _, err := os.Stat(pathName); os.IsNotExist(err) {
+		return nil, fmt.Errorf("provided device/file %s does not exist", pathName)
+	}
+
+	openMode := os.O_RDONLY
+
+	if !isReadonly {
+		openMode |= os.O_RDWR | os.O_EXCL
+	}
+
+	f, err := os.OpenFile(pathName, openMode, 0o600)
+	if err != nil {
+		return nil, fmt.Errorf("could not open device %s with mode %v: %w", pathName, openMode, err)
+	}
+
+	return rawBackend{
+		storage:  f,
+		readonly: isReadonly,
+	}, nil
+}
+
+// Create a backend.Storage from a path to an image file.
+// Should pass a path to a file /tmp/foo.img
+// The provided file must not exist at the time you call CreateFromPath()
+func CreateFromPath(pathName string, size int64) (backend.Storage, error) {
```

Comment:
See the comment above. If we have `Open()` with mode, then `Create()` does not need to exist, or at most is a wrapper around `Open(os.O_CREATE)`. Or do we think it is simpler this way?

Reply:
I've separated them intentionally to not overwrite existing image. Open just opens and Create, ahem, creates! )) In no chance existing file will be truncated. But I agree, it looks like some sugar around lower level functions.

What would be your final thoughts on this?

Reply:
As you probably can tell by now, my lower-level thoughts are mutually contradictory. 😆

I think for now you are right, we should leave it with two distinct functions: `Open()` and `Create()`.

Reply:
So, we're leaving it as it is now?

Resolving this to reduce noise in discussion threads.